core/src/main/java/com/graphhopper/routing/ProfileResolver.java

```diff
@@ -200,10 +198,10 @@ private String getCHRequestAsString(HintsMap hintsMap, Boolean edgeBased, Intege
     }
 
     private Boolean getEdgeBased(HintsMap hintsMap) {
-        return hintsMap.has(Parameters.Routing.EDGE_BASED) ? hintsMap.getBool(Parameters.Routing.EDGE_BASED, false) : null;
+        return (Boolean) hintsMap.getObject(Parameters.Routing.EDGE_BASED, null);
     }
 
     private Integer getUTurnCosts(HintsMap hintsMap) {
-        return hintsMap.has(Parameters.Routing.U_TURN_COSTS) ? hintsMap.getInt(Parameters.Routing.U_TURN_COSTS, INFINITE_U_TURN_COSTS) : null;
+        return (Integer) hintsMap.getObject(Parameters.Routing.U_TURN_COSTS, null);
```

Comment:
Does this not give a NPI in case there is no u_turn_costs entry?

Reply:
If there is no u_turn_cost entry it will return null and casting this to Integer is possible without NPE. So it should return `null` like before.

Reply:
I thought casting null gives NPE, but you are right.

Reply:
There is another problem with this approach: If the value in `hintsMap` is a String (like `"true"`) this will lead to a `ClassCastException`.

Reply:
... and the reason I ran into this is that `RouteResource#translateTurnCostsParamToEdgeBased` is not using `Helper.toObject`... This of course can be fixed, but I still think the old approach is better (because it avoids the `ClassCastException`). wdyt?

Reply:
For example I could do a query with `edge_based=chocolate` and this will yield a `ClassCastException` (a 500 error)

Reply:
Oh, this is very ugly. And this is a problem we now have for all methods. E.g. `short_fastest.distance_factor=blup` will cause the same ClassCastException.

Should we modify the getter methods and if not the expected class => return the default?

Reply:
Are you sure? `hints.getFloat("short_fastest.distance_factor", 3.6)` should just work. The problem here is only that we use `getObject` and then cast to Boolean?

Reply:
> Should we modify the getter methods and if not the expected class we return the default?

Isn't this exactly what we do?

Reply:
Oh, sorry. Again a stale-state in my mind. I had a cast approach for PMap before I realized that instanceof is better...